homeassistant/components/bsblan/__init__.py

```diff
@@ -35,22 +34,27 @@ class HomeAssistantBSBLANData:
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up BSB-Lan from a config entry."""
 
-    session = async_get_clientsession(hass)
-    bsblan = BSBLAN(
-        entry.data[CONF_HOST],
+    # create config using BSBLANConfig
+    config = BSBLANConfig(
+        host=entry.data[CONF_HOST],
         passkey=entry.data[CONF_PASSKEY],
         port=entry.data[CONF_PORT],
         username=entry.data.get(CONF_USERNAME),
         password=entry.data.get(CONF_PASSWORD),
-        session=session,
```

Comment:
Why do we remove the session injector?

Reply:
I thought it was not needed as bsblan now created its own session. But I guess this is not correct?

Reply:
The session we have is a bit more optimized and it saves us from creating a client session for every integration